flash/data/process.py

```diff
@@ -128,7 +133,31 @@ def validating(self, val: bool) -> None:
             self._running_stage = None
 
 
-class Preprocess(Properties, Module):
+class AbstractPreprocess(ABC):
+
+    @abstractstaticmethod
+    def version() -> str:
+        """
+        Override to versioned to version your preprocess.
+        """
+        pass
+
+    @abstractmethod
```

Comment:
Why not rely on the `nn.Module` `state_dict`/`load_state_dict` for this?

Preprocess does inherit it

Reply:
Currently, state_dict from nn.Module calls Preprocess `_save_to_state_dict` nn.Module hook which calls the users `save_state_dict` hook which is mandatory as `abstractmethod`.

By doing so, users are force to properly serialise their Preprocess objects.